main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/bitrise-tools/go-steputils/tools"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+
+	"github.com/bitrise-io/go-utils/command/git"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+)
+
+type config struct {
+	Version   string `env:"version,required"`
+	Overwrite bool   `env:"overwrite,required"`
+}
+
+func failf(msg string, args ...interface{}) {
+	log.Errorf(msg, args...)
+	os.Exit(1)
+}
+
+func main() {
+	var cfg config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Issue with input: %s", err)
+	}
+	stepconf.Print(cfg)
+
+	isInstalled := true
+	if _, err := exec.LookPath("flutter"); err != nil {
+		isInstalled = false
+	}
+
+	if cfg.Overwrite || !isInstalled {
```

Comment:
i would check if the current flutter version matches to the desired version (i would only check if the desired state is a tag, this way we can match version, if branch is provided, just skip this check)